scripts/irods/upgrade_configuration.py

```diff
@@ -92,70 +102,60 @@ def upgrade(irods_config):
-                            'address_type': e['address_type'],
-                            'addresses': [o['address'] for o in e['addresses']]
-                        })
+            if 'host_entries' in hosts_config and len(hosts_config['host_entries']) > 0:
+                # Ignores the schema information in the JSON file.
+                # Convert each addresses object into a list of address strings.
+                for e in hosts_config['host_entries']:
+                    write_changes_to_disk = True
+                    server_config['host_resolution']['host_entries'].append({
+                        'address_type': e['address_type'],
+                        'addresses': [o['address'] for o in e['addresses']]
+                    })
 
         #
         # host_access_control_config.json
         #
 
-        merged_host_access_control_config = False
-
-        # Merge the host_access_control_config.json file into server_config.json only if the
-        # server_config.json does not contain the host_access_control property.
-        if 'host_access_control' not in server_config:
-            merged_host_access_control_config = True
+        # Merge host_access_control_config.json into server_config.json if it is available.
+        if os.path.exists(paths.host_access_control_config_path()):
+            with open(paths.host_access_control_config_path()) as f:
+                host_access_control = json.load(f)
 
-            # Set default values for the host_access_control.
-            server_config['host_access_control'] = {'access_entries': []}
+            if 'access_entries' in host_access_control and len(host_access_control['access_entries']) > 0:
+                write_changes_to_disk = True
+                # Ignores the schema information in the JSON file.
+                server_config['host_access_control']['access_entries'] = host_access_control['access_entries']
 
-            # Merge host_access_control_config.json into server_config.json if it is available.
-            if os.path.exists(paths.host_access_control_config_path()):
-                with open(paths.host_access_control_config_path()) as f:
-                    host_access_control = json.load(f)
-
-                if 'access_entries' in host_access_control and len(host_access_control['access_entries']) > 0:
-                    # Ignores the schema information in the JSON file.
-                    server_config['host_access_control']['access_entries'] = host_access_control['access_entries']
-
-        # Write changes to disk.
-        if merged_hosts_config or merged_host_access_control_config:
+        if write_changes_to_disk:
             irods_config.commit(server_config, paths.server_config_path(), make_backup=True)
 
+        # TODO Should we rename hosts_config.json and host_access_control_config.json so that
+        # the upgrade process does not process them anymore?
```

Comment:
Thoughts?

Reply:
I think upgrade should sniff if those stanzas/values already exist in server_config.json and do the conversion if not found.  I don't think we should rename or delete these other two 'legacy' files?

Reply:
Renaming them could be elegant... perhaps appending ... `.absorbed_by_4.3.0`?   Hmm....

But definitely, they'd be ignored.

Reply:
Thinking about it more, how about `.absorbed_by_4.3.0_server_config`?

Reply:
I've adjusted the code to append `.absorbed_by_4.3.0_server_config` instead of the former because it is more descriptive about what happened.